From 9d11dccdc7dacf5cb759015f40dc6222a002163d Mon Sep 17 00:00:00 2001
From: haiqiang yao <haiqiangx.yao@intel.com>
Date: Wed, 16 Mar 2016 17:01:23 +0800
Subject: [PATCH 4/5] atomisp2:Add tdf parameter for isp2.7 pipe

this patch will add tdf parameter for isp2.7 pipe.

Depends on:https://icggerrit.ir.intel.com/#/c/107991/1
Depends on:https://icggerrit.ir.intel.com/#/c/105955/

Change-Id: Icecb4f4d417e0c8585dc1e73d1d95400852f36ce
Original-Tracked_On: #H1504179048
Signed-off-by: haiqiang yao <haiqiangx.yao@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/108108
Approver: Bin B Han <bin.b.han@intel.com>
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Build: Dessert M BXT RVP (SC)
Integrator: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-25608
Signed-off-by: Tao Wei <wei.tao@intel.com>
Reviewed-on: https://android.intel.com:443/483551
---
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c | 13 +++++++++++++
 .../pci/atomisp2/atomisp_driver/atomisp_compat_css20.c      |  6 ++++++
 .../pci/atomisp2/atomisp_driver/atomisp_compat_css20.h      |  4 ++++
 .../pci/atomisp2/atomisp_driver/atomisp_compat_ioctl32.h    |  1 +
 .../media/pci/atomisp2/atomisp_driver/atomisp_subdev.h      |  1 +
 5 files changed, 25 insertions(+)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
index 97279f0..7ab93d3 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
@@ -3333,6 +3333,9 @@ void atomisp_apply_css_parameters(
 	if (css_param->update_flag.bnlm_config)
 		atomisp_css_set_bnlm_config(asd, &css_param->bnlm_config);
 
+	if (css_param->update_flag.tdf_config)
+		atomisp_css_set_tdf_config(asd, &css_param->tdf_config);
+
 	atomisp_css_set_isp_config_id(asd, css_param->isp_config_id);
 	/*
 	 * These configurations are on used by ISP1.x, not for ISP2.x,
@@ -3722,6 +3725,16 @@ int atomisp_cp_general_isp_parameters(struct atomisp_sub_device *asd,
 			(struct atomisp_bnlm_config *) &css_param->bnlm_config;
 	}
 
+	if (arg->tdf_config && (from_user || !cur_config->tdf_config)) {
+		if (copy_from_compatible(&css_param->tdf_config,
+				arg->tdf_config,
+				sizeof(struct atomisp_css_tdf_config),
+				from_user))
+			return -EFAULT;
+		css_param->update_flag.tdf_config =
+			(struct atomisp_tdf_config *) &css_param->tdf_config;
+	}
+
 	if (from_user)
 		css_param->isp_config_id = arg->isp_config_id;
 	/*
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
index 033b120..b5d0c5a 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
@@ -3963,6 +3963,12 @@ void atomisp_css_set_bnlm_config(struct atomisp_sub_device *asd,
 	asd->params.config.bnlm_config = bnlm_config;
 }
 
+void atomisp_css_set_tdf_config(struct atomisp_sub_device *asd,
+			struct atomisp_css_tdf_config *tdf_config)
+{
+	asd->params.config.tdf_config = tdf_config;
+}
+
 int atomisp_css_get_dp_config(struct atomisp_sub_device *asd,
 			struct atomisp_dp_config *config)
 {
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.h
index 84005ac..036ab39 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.h
@@ -94,6 +94,7 @@
 #define atomisp_css_macc1_5_table	ia_css_macc1_5_table
 #define atomisp_css_xnr3_0_11_config	ia_css_xnr3_0_11_config
 #define atomisp_css_bnlm_config	ia_css_bnlm_config
+#define atomisp_css_tdf_config ia_css_tdf_config
 
 
 #define CSS_PIPE_ID_PREVIEW	IA_CSS_PIPE_ID_PREVIEW
@@ -298,6 +299,9 @@ void atomisp_css_set_xnr3_0_11_config(struct atomisp_sub_device *asd,
 void atomisp_css_set_bnlm_config(struct atomisp_sub_device *asd,
 			struct atomisp_css_bnlm_config *bnlm_config);
 
+void atomisp_css_set_tdf_config(struct atomisp_sub_device *asd,
+			struct atomisp_css_tdf_config *tdf_config);
+
 unsigned int atomisp_css_debug_get_dtrace_level(void);
 
 int atomisp_css_debug_dump_isp_binary(void);
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_ioctl32.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_ioctl32.h
index 2fd74a4..dfaa8b9 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_ioctl32.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_ioctl32.h
@@ -240,6 +240,7 @@ struct atomisp_parameters32 {
 	compat_uptr_t macc1_5_table;
 	compat_uptr_t xnr3_0_11_config;
 	compat_uptr_t bnlm_config;
+	compat_uptr_t tdf_config;
 
 	compat_uptr_t	lin_2500_config;       /* Skylake: Linearization config */
 	compat_uptr_t	obgrid_2500_config;    /* Skylake: OBGRID config */
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.h
index 8574e03..4a59d7d 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.h
@@ -199,6 +199,7 @@ struct atomisp_css_params {
 	struct ia_css_macc1_5_table macc1_5_table;
 	struct ia_css_xnr3_0_11_config xnr3_0_11_config;
 	struct ia_css_bnlm_config bnlm_config;
+	struct ia_css_tdf_config tdf_config;
 
 
 	/*
-- 
1.9.1

